src/Compilers/CSharp/Portable/Utilities/ValueSetFactory.EnumeratedValueSetFactory.cs

```diff
@@ -37,6 +37,13 @@ public IValueSet<T> Related(BinaryOperatorKind relation, T value)
             IValueSet IValueSetFactory.Related(BinaryOperatorKind relation, ConstantValue value) =>
                 value.IsBad ? EnumeratedValueSet<T, TTC>.AllValues : this.Related(relation, default(TTC).FromConstantValue(value));
 
+            bool IValueSetFactory.Related(BinaryOperatorKind relation, ConstantValue left, ConstantValue right)
+            {
+                Debug.Assert(relation == BinaryOperatorKind.Equal);
+                TTC tc = default;
+                return tc.FromConstantValue(left)!.Equals(tc.FromConstantValue(right));
```

Comment:
Can we add a constraint so that we can avoid the suppression here?

Reply:
We cannot add a struct constraint, because this is used for `string`.  But I can make this an assertion.

---
In reply to: [400457145](https://github.com/dotnet/roslyn/pull/42313#discussion_r400457145) [](ancestors = 400457145)

Reply:
Could we use a `notnull` constraint?

Reply:
Yes.

---
In reply to: [400558934](https://github.com/dotnet/roslyn/pull/42313#discussion_r400558934) [](ancestors = 400558934)